src/controller/SetUpCodePairer.cpp

```diff
@@ -161,8 +161,35 @@ void SetUpCodePairer::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ERROR
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 #if CHIP_DEVICE_CONFIG_ENABLE_DNSSD
-void SetUpCodePairer::OnDiscoveredDevice(const Dnssd::DiscoveredNodeData & nodeData)
+
+bool SetUpCodePairer::NodeMatchesCurrentFilter(const Dnssd::DiscoveredNodeData & nodeData)
 {
+    switch (currentFilter.type)
+    {
+    case Dnssd::DiscoveryFilterType::kShort:
+        return (nodeData.longDiscriminator >> 8) == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kLong:
+        return nodeData.longDiscriminator == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kVendor:
+        return nodeData.vendorId == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kDeviceType:
+        return nodeData.deviceType == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kCommissioningMode:
+        return nodeData.commissioningMode == currentFilter.code;
+    case Dnssd::DiscoveryFilterType::kNone:
+    case Dnssd::DiscoveryFilterType::kInstanceName:
+    case Dnssd::DiscoveryFilterType::kCommissioner:
+    case Dnssd::DiscoveryFilterType::kCompressedFabricId:
+        return false;
+    }
+    return false;
+}
+void SetUpCodePairer::NotifyCommissionableDeviceDiscovered(const Dnssd::DiscoveredNodeData & nodeData)
+{
+    if (!NodeMatchesCurrentFilter(nodeData))
```

Comment:
Is this too strict overall? Couldn't you have multiple filters apply at once?

Reply:
In theory, yes, but the current code assumes one running  at a time. This PR doesn't change that.